document current wasi thread support (#1230)
MicroEJ/docs: ApplicationDeveloperGuide/managedc/wasi.rst

   .. code:: java

      package com.mycompany;

      import ej.wasm.WasmFunction;
      import ej.wasm.WasmMemory;
      import ej.wasm.WasmModule;
      import ej.wasi.*;
      import ej.wasi.thread.WasiThreadRedirection;

      import java.io.IOException;

      @WasmModule("my_app")
      public class MyApp {

          @WasmMemory
          private static byte[] Memory;

          // Define the binding to 'wasi_thread_start' function
          @WasmFunction
          private static final native void wasi_thread_start(int threadId, int args);

          public static void main(String[] args) {

              // Initialize WASI with the thread start redirection
              WasiThreadRedirection wtr = new WasiThreadRedirection() {
                  public void start(int threadId, int args) {
                      wasi_thread_start(threadId, args);
                  }
              };

              try {
                  Wasi.init(Memory, wtr);
              } catch (IOException e) {
                  // error during initialization
              }

              // ...
          }
      }

> Limitations
> ~~~~~~~~~~~

Reviewer: ```suggestion
WASI Threads Limitations
~~~~~~~~~~~~~~~~~~~~~~~~
```

An Application can declare at most one Wasm module.

> Stack Management

Reviewer: I would add this section as a paragraph in the "Limitations" section.

~~~~~~~~~~~~~~~~

> The Java ``synchronized`` keyword is no longer mandatory on methods bounds to C functions, since multiple threads can now re-enter C code. 

Reviewer: ```suggestion
The Java ``synchronized`` keyword is no longer mandatory on methods bound to C functions, as multiple threads can now invoke C code. 
```

However, threads created from Java still share the main stack of linear memory, so you must ensure they do not execute C code simultaneously.

> Threads created from Managed C (pthreads) have a dedicated stack dynamically allocated in the heap of linear memory. 

Reviewer: ```suggestion
In contrast, pthreads created in Managed C have their own stack, which is dynamically allocated in linear memory, allowing them to execute C code in parallel without restriction.
```
